tools/patch/grammar.patch

```diff
@@ -0,0 +1,1005 @@
+diff --git a/Grammar/python.gram b/Grammar/python.gram
+index 64e205e7fd..2f20f2655b 100644
+--- a/Grammar/python.gram
++++ b/Grammar/python.gram
+@@ -2,7 +2,7 @@
+ 
+ @trailer '''
+ void *
+-_PyPegen_parse(Parser *p)
++pegen.parse(Parser *p)
+ {
+     // Initialize keywords
+     p->keywords = reserved_keywords;
+@@ -27,51 +27,51 @@ _PyPegen_parse(Parser *p)
+ 
+ // The end
+ '''
+-file[mod_ty]: a=[statements] ENDMARKER { _PyPegen_make_module(p, a) }
+-interactive[mod_ty]: a=statement_newline { Interactive(a, p->arena) }
+-eval[mod_ty]: a=expressions NEWLINE* ENDMARKER { Expression(a, p->arena) }
+-func_type[mod_ty]: '(' a=[type_expressions] ')' '->' b=expression NEWLINE* ENDMARKER { FunctionType(a, b, p->arena) }
+-fstring[expr_ty]: star_expressions
++file[mod]: a=[statements] ENDMARKER { pegen.make_module(this, a) }
++interactive[mod]: a=statement_newline { new astnodes.Interactive(a) }
++eval[mod]: a=expressions NEWLINE* ENDMARKER { new astnodes.Expression(a) }
++func_type[mod]: '(' a=[type_expressions] ')' '->' b=expression NEWLINE* ENDMARKER { new astnodes.FunctionType(a, b) }
++fstring[expr]: star_expressions
+ 
+ # type_expressions allow */** but ignore them
+-type_expressions[asdl_seq*]:
++type_expressions[any]:
```

Comment:
having a manual curated grammar file might help us with getting better types for these cases because automatically it would be hard to figure out what is actually in the seq.

problem is we can't have `[]`in the `[]` so we'll need to think of something there.

Reply:
agreed - we can manually put in specific types as we get to grips with each rule. If we want to to have `[]` we can maybe use the `*` and clean the type in the parser generator.
Hopefully when remove `// @ts-nocheck` from the generated parser typescript will just tell us what these should be 😉

Reply:
that would be amazing 😄 but I don't think it'll be that simple 😂